### DISCUSSION TITLE
Query on Running DNNMark Without CUDA - CPU-Only Simulation Feasibility?

### DISCUSSION THREAD

**Discussion by EricChenSudo** [gem5/gem5#659] [Q&A] [answered]
Hello authors,
Is it possible to run gem5-GPU DNNMark simulations effectively on a CPU-only server?

Thank you in advance for your time and assistance. Looking forward to your responses!

**Comment by abmerop** [comment]
Hello,

If I understand the question correctly, you are wondering about the simulation speed of gem5 simulating GPU applications?  If so, DNNMark is quite fast.  I believe it is/was part of the weekly or maybe daily tests.

I don't know too much about the suite, but the default inputs are relatively small and I assume you can scale them up as much as you want.

If you are trying to *compile* DNNMark to *run* on a CPU, I don't know if it has that option.  If you are dedicated on doing that you can try to use the HIP-CPU framework: https://github.com/ROCm-Developer-Tools/HIP-CPU .  However, this does not use the ROCm runtime (I believe) so you will likely run into issues with dependencies like MIOpen.

@mattsinc may have more advice too.

**Reply by mattsinc** [reply, marked as answer]
DNNMark itself only supports running on AMD and NVIDIA GPUs: https://github.com/shidong-ai/DNNMark

So, no, by default, DNNMark cannot run its kernels on the CPU in gem5.  But that has nothing to do with gem5 support, it's just DNNMark's support.  I suppose in theory you could add a backend that supports X86 BLAS libraries or something, but it does not exist today.

@abmerop is right that you could also try HIP-CPU to convert DNNMark to run on a CPU (e.g., in gem5).  But like Matt P said that is not perfect and likely will summarize the workload a lot more than you are shooting for here.

**Comment by EricChenSudo** [comment]
Hello,

Thank you very much for your response. I appreciate the information about DNNMark and its performance.

However, my question was specifically about the feasibility of simulating gem5-GPU DNNMark (especially the  fully-connected (fc) model) on a server that only has CPUs, without a GPU. To clarify, I am referring to running these simulations using the actual CPUs of the server, not the virtual CPUs within gem5 (virtual hardware of gem5 in this simulation has CPU and GPU). Is it possible to conduct these simulations effectively on a CPU-only server, or is a GPU essential for this purpose?

Thank you once again for your assistance and any further insights you can provide on this matter.

**Reply by abmerop** [reply]
gem5 only runs on CPUs.  You don't need a GPU or any GPU software installed at all.  Docker images are provided in case you need to compile GPU applications to run in gem5.

**Reply by mattsinc** [reply]
Right, you don't need any kind of GPU on your machine to run gem5.
